src/controllers/vstatefulset_controller/exec/reconciler.rs

```diff
@@ -0,0 +1,1570 @@
+            vsts@,
+            resp_o.deep_view(),
+            state@,
+        ),
+{
+    let ordinal_or_none = get_largest_ordinal_of_unmatched_pods(vsts, &state.needed);
+    if ordinal_or_none.is_some() {
+        let ordinal = ordinal_or_none.unwrap();
+
+        proof {
+            let pods = state.needed.deep_view();
+            let ordinals = Seq::new(pods.len(), |i: int| i as nat);
+            let pred = |ordinal: nat| pods[ordinal as int] is Some && !model_reconciler::pod_matches(vsts@, pods[ordinal as int]->0);
+            let filtered = ordinals.filter(pred);
+            assert(filtered.contains(ordinal as nat));
+            seq_filter_contains_implies_seq_contains(ordinals, pred, ordinal as nat);
+            assert(ordinals.contains(ordinal as nat));
+        }
+
+        let pod = state.needed[ordinal as usize].clone().unwrap();
+
+        if pod.metadata().name().is_none() {
+            return (error_state(state), None)
+        }
+
+        let req = KubeAPIRequest::GetThenDeleteRequest(KubeGetThenDeleteRequest {
+            api_resource: Pod::api_resource(),
+            name: pod.metadata().name().unwrap(),
+            namespace: vsts.metadata().namespace().unwrap(),
+            owner_ref: vsts.controller_owner_ref(),
+        });
+        let state_prime = VStatefulSetReconcileState {
+            reconcile_step: VStatefulSetReconcileStep::AfterDeleteOutdated,
+            ..state
+        };
+        (state_prime, Some(Request::KRequest(req)))
+    } else {
+        (done_state(state), None)
+    }
+}
+
+pub fn handle_after_delete_outdated(
+    vsts: &VStatefulSet,
+    resp_o: Option<Response<VoidEResp>>,
+    state: VStatefulSetReconcileState,
+) -> (res: (VStatefulSetReconcileState, Option<Request<VoidEReq>>))
+    requires
+        vsts@.well_formed(),
+    ensures
+        (res.0@, res.1.deep_view()) == model_reconciler::handle_after_delete_outdated(
+            vsts@,
+            resp_o.deep_view(),
+            state@,
+        ),
+{
+    let res = is_some_k_get_then_delete_resp!(resp_o);
+    proof {
+        assert(res == is_some_k_get_then_delete_resp_view(resp_o.deep_view()));
+    }
+    if is_some_k_get_then_delete_resp!(resp_o) {
+        let result = extract_some_k_get_then_delete_resp!(resp_o);
+        if result.is_ok() {
+            (done_state(state), None)
+        } else {
+            (error_state(state), None)
+        }
+    } else {
+        // This should be unreachable
+        (error_state(state), None)
+    }
+}
+
+pub fn handle_after_create_or_skip_pvc_helper(
+    vsts: &VStatefulSet,
+    state: VStatefulSetReconcileState,
+) -> (res: (VStatefulSetReconcileState, Option<Request<VoidEReq>>))
+    requires
+        vsts@.well_formed(),
+        state.pvc_index < usize::MAX,
+    ensures
+        (res.0@, res.1.deep_view()) == model_reconciler::handle_after_create_or_skip_pvc_helper(
+            vsts@,
+            state@,
+        ),
```

Comment:
As a sanity check, can we add a postcondition here to assert that state_prime.pvc_index is also in the bound? Similar for other functions with + 1.